workbench-core/datasets/src/wbcDataSetsAuthorizationPlugin.ts

```diff
@@ -68,21 +68,18 @@ export class WbcDataSetsAuthorizationPlugin implements DataSetsAuthorizationPlug
         identityPermissions
       });
 
-    const newPermissions: DataSetPermission[] = createdPermission.data.identityPermissions.map((i) => {
-      const accessLevel: DataSetsAccessLevel = i.action === 'READ' ? 'read-only' : 'read-write';
-      return {
-        identity: i.identityId,
-        identityType: i.identityType,
-        accessLevel: accessLevel
-      };
-    });
+    const permissions: PermissionsResponse[] = this._identityPermissionsToPermissionsResponse(
+      createdPermission.data.identityPermissions
+    );
+    const permissionsCount = permissions.length;
 
-    return {
-      data: {
-        dataSetId: createdPermission.data.identityPermissions[0].subjectId,
-        permissions: newPermissions
-      }
-    };
+    if (permissionsCount !== 1) {
+      throw new InvalidPermissionError(
```

Comment:
when we expect this error, are we expecting users to delete those permissions?

Reply:
we are not expecting this error.